src/value.rs

```diff
@@ -848,7 +848,7 @@ impl ValueConstructible for StructuralValue {
         bound: NonZeroPow2Usize,
     ) -> Self {
         let elements: Vec<Self> = elements.into_iter().collect();
-        assert!(bound.get() <= elements.len(), "Too many elements");
+        assert!(elements.len() < bound.get(), "Too many elements");
```

Comment:
In 3d8a00854e31f88573df7687e6187d99c231dacc:

I struggled to understand this commit. If you mean to assert that there are strictly fewer elements than the bound (which does sound right) then you need to change the error message in the `assert`.

The commit message also says that this code is called "after another assertion that checks the correct invariant", which I *think* refers to line 746 which `debug_assert`s equality between `elements.len` and some other quantity. But I don't see what that assertion has to do with this one.

Maybe you are referring to the assertion in `Partition::from_slice`? That assertion appears to be equivalent to this new one, which means that the previous code should never have been possible to call! What's going on here?

Reply:
`<StructuralValue as ValueConstructible>::list` is called by `<StructuralValue as From<&'a Value>::from`. The latter has the assertion `debug_assert!(elements.len() < bound.get())`. I will update the assert message. "Too many elements" is meant to be displayed for the error case, but I can see how a statement about the should-be state is clearer.

Reply:
I find with assert messages it's clearest to use "should" language. "There should be strictly fewer elements than the bound". Also in this case we should put the exact numbers into the error messages (since the `assert!` macro can't provide them itself as it's just asserting a boolean).

Reply:
Cool, I'm happy with this. Though still a little mystified that the original code ever worked. I think it must have never been called.